Whiskey/Functions/Install-WhiskeyDotNetTool.ps1

```diff
@@ -42,93 +43,114 @@ function Install-WhiskeyDotNetTool
+                return 'dotnet'
+            }
         }
 
-        $globalJsonSdkOptions =
-            $globalJson |
-            Select-Object -ExpandProperty 'sdk' -ErrorAction Ignore
-
-        $globalJsonVersion =
-            $globalJsonSdkOptions |
-            Select-Object -ExpandProperty 'version' -ErrorAction Ignore
-
-        $globalJsonRollForward =
-            $globalJsonSdkOptions |
-            Select-Object -ExpandProperty 'rollForward' -ErrorAction Ignore
-
-
-        $rollForwardStrategy = [Whiskey.DotNetSdkRollForward]::Disable
-        $validRollForwardValues = [Whiskey.DotNetSdkRollForward].GetEnumNames()
-        if ($globalJsonRollForward -notin $validRollForwardValues)
+        $globalJsonPath = Join-Path -Path $WorkingDirectory -ChildPath 'global.json'
+        if ( -not (Test-Path -Path $globalJsonPath -PathType Leaf) )
         {
-            $msg = "sdk.rollForward value ""$($globalJsonRollForward)"" in ""$($globalJsonPath)"" is not " +
-            "one of the valid .NET SDK roll forward strategies: $($validRollForwardValues -join ', ')." +
-            "$(([Environment]::NewLine)*2)If ""$($globalJsonRollForward)"" is a new valid .NET SDK roll forward strategy " +
-            "then submit a Whiskey issue (https://github.com/webmd-health-services/Whiskey/issues) asking us to add " +
-            "support for it.$(([Environment]::NewLine)*2)Defaulting roll forward strategy to ""$($rollForwardStrategy)""."
-            Write-Warning -Message $msg
+            $globalJsonPath = Join-Path -Path $InstallRoot -ChildPath 'global.json'
         }
-        else
+
+        $sdkVersion = $null
+        if ( Test-Path -Path $globalJsonPath -PathType Leaf )
         {
-            $rollForwardStrategy = [Whiskey.DotNetSdkRollForward] $globalJsonRollForward
-        }
+            try
+            {
+                $globalJson = Get-Content -Path $globalJsonPath -Raw | ConvertFrom-Json
+            }
+            catch
+            {
+                $msg = "Failed to install .NET because global.json file ""$($globalJsonPath)"" contains invalid JSON."
+                Write-WhiskeyError -Message $msg
+                return
+            }
+
+            $globalJsonSdkOptions =
+                $globalJson |
+                Select-Object -ExpandProperty 'sdk' -ErrorAction Ignore
+
+            $globalJsonVersion =
+                $globalJsonSdkOptions |
+                Select-Object -ExpandProperty 'version' -ErrorAction Ignore
+
+            $globalJsonRollForward =
+                $globalJsonSdkOptions |
+                Select-Object -ExpandProperty 'rollForward' -ErrorAction Ignore
+
+            $rollForwardStrategy = [WhiskeyDotNetSdkRollForward]::Disable
+            $validRollForwardValues = [WhiskeyDotNetSdkRollForward].GetEnumNames()
+
+            if ($globalJsonRollForward)
+            {
+                if ($globalJsonRollForward -and $globalJsonRollForward -notin $validRollForwardValues)

```

Comment:
You can remove `$globalJsonRollForward -and` as it has already been evaluated just above.

Reply:
🙃